snapcraft/tests/matchers.py

```diff
@@ -18,16 +18,39 @@
 import testtools
 
 
-class HasArchitecture:
-    """Match if the file was built for the expected architecture"""
-
-    def __init__(self, expected_arch):
-        self._expected_arch = expected_arch
+class HasBinaryFileHeader:
```

Comment:
This should probably start with an underscore seeing that it can't be used standalone.

Reply:
I changed it so it can be used on its own instead. Also made the subclasses smaller.